### PR TITLE
!fix(missing-kind-type): convert auto generated enums to union types

### LINKED ISSUE #134: Missing type on OrderQuoteSide.kind
According to your API documentation [here](https://docs.cow.fi/cow-sdk/sign-and-post-orders/step-1-get-market-price). I could use in the `getQuote` function the `OrderKind.SELL` or `OrderKind.BUY`.

But, it is impossible because `kind` only accepts `OrderQuoteSide` enum, as in the following image. 
image

Then, I have to use `OrderQuoteSide.kind` instead `OrderKind`. 

## And there is the second issue
`OrderQuoteSide.kind` has only `SELL` enum, not the buy one. So, it's impossible to use `OrderQuoteSide.kind.BUY`. As the images below:
image
image

> cow-sdk/dist/order-book/generated/models/OrderQuoteSide.d.ts#L2

I'm using the `@cowprotocol/cow-sdk@^2.3.0-rc.7`
I tried to help with a [PR](https://github.com/cowprotocol/cow-sdk/pull/133), but @mfw78 told me that is not the right place to change it. Feel free to refuse that.

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Closes #134 

Converting openapi enums to union types

# ⚠️ Attention!! ⚠️ 

This is a breaking change and will have cascading effects on consumers, as the enums will be removed!

# Context

We auto-generate the types based on the orderbook/openapi spec.
The OrderQuoteSide is defined [here](https://github.com/cowprotocol/services/blob/main/crates/orderbook/openapi.yml#L1153)
It uses kind.sell for some options and kind.buy for another.
We use openapi[ cli tool to generate the corresponding TS types](https://github.com/cowprotocol/cow-sdk/blob/main/package.json#L29C42-L29C130)
The resulting type can be seen [here](https://unpkg.com/browse/@cowprotocol/cow-sdk@2.3.0/dist/order-book/generated/models/OrderQuoteSide.d.ts)

It does NOT include the enum `buy`.

It's a limitation of the lib used for generating the types. See [this issue](https://github.com/ferdikoomen/openapi-typescript-codegen/issues/1424).

This change applies the suggested workaround: 
> Use union types instead of enums (often though it's desirable to have the options available at runtime)

[Here's a write up](https://github.com/ferdikoomen/openapi-typescript-codegen/blob/HEAD/docs/enum-vs-union-types.md) about this option.

# Test

- Unit tests
- Checked the generated file contains the missing type -> https://github.com/cowprotocol/cow-sdk/blob/bda19dd8b2d14843b1305ad7787f540950c482c4/src/order-book/generated/models/OrderQuoteSide.ts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "prepare": "npm run build",
     "prepublishOnly": "npm test && npm run lint",
     "graphql:codegen": "graphql-codegen --config graphql-codegen.yml",
-    "swagger:codegen": " openapi --input https://raw.githubusercontent.com/cowprotocol/services/main/crates/orderbook/openapi.yml --output src/order-book/generated --exportServices false --exportCore false"
+    "swagger:codegen": "openapi --input https://raw.githubusercontent.com/cowprotocol/services/main/crates/orderbook/openapi.yml --output src/order-book/generated --exportServices false --exportCore false --useUnionTypes"
```
Comment (alfetopito) [inline package.json:29]: This is the main change. See what the option does [in the docs](https://github.com/ferdikoomen/openapi-typescript-codegen/blob/HEAD/docs/enum-vs-union-types.md)

Everything else is adapting to the consequences of it.

**Review by anxolin — COMMENTED** [review]
Oh no! do we really need to do this? 

We've been using the enums with no issues before.

**Comment by alfetopito** [comment]
> Is the options 2 possible?
> 
> `2. Restructure the schema and split it up into further components to avoid conflicts`?

It requires changes to the openapi specs in the services repo instead.
I can also try that as an alternative solution.

**Comment by alfetopito** [comment]
Attempt #2, which requires services repo changes:
- https://github.com/cowprotocol/services/pull/1732
- https://github.com/cowprotocol/cow-sdk/pull/138

**Pull request closed without merging**